core/src/main/scala/com/databricks/labs/remorph/generators/sql/LogicalPlanGenerator.scala

```diff
@@ -392,16 +400,26 @@ class LogicalPlanGenerator(
 
   private def subQueryAlias(ctx: GeneratorContext, subQAlias: ir.SubqueryAlias): String = {
     val subquery = subQAlias.child match {
-      case l: ir.Lateral => lateral(ctx, l)
+      case l: ir.Lateral => lateralView(ctx, l)
       case _ => s"(${generate(ctx, subQAlias.child)})"
     }
     val tableName = expr.generate(ctx, subQAlias.alias)
-    val table = if (subQAlias.columnNames.isEmpty) {
-      tableName
-    } else {
-      tableName + subQAlias.columnNames.map(expr.generate(ctx, _)).mkString("(", ", ", ")")
-    }
-    s"$subquery AS $table"
+
+    val table =
+      if (subQAlias.columnNames.isEmpty) {
+        s"AS $tableName"
+      } else {
+        // Added this to handle the case for POS Explode
+        subQAlias.columnNames match {
+          case Seq(ir.Id("value", _), ir.Id("index", _)) =>
+            val columnNamesStr = subQAlias.columnNames.reverse.map(expr.generate(ctx, _)).mkString(",")
+            s"$tableName AS $columnNamesStr"
+          case _ =>
+            val columnNamesStr = tableName + subQAlias.columnNames.map(expr.generate(ctx, _)).mkString("(", ", ", ")")
+            s"AS $columnNamesStr"
+        }
```

Comment:
We are overly specific here. For example, if column names come in a different order (`index` coming before `value`) or if there are other column names involved, the case on line 414 wouldn't be matched and we'd end up producing the wrong output.

What we should do instead is matching on `subQAlias.child` as we do above. In case of a `Lateral`, we'd output the alias in the form `tableName AS col1, col2, col3` and otherwise we'd output the alias in the form `AS tableName(col1, col2, col3)`.

Reply:
done. as discussed but these are columns which we are adding as an alias to pos explode default columns and yes it can occur in any order, added the fix please take a look